### PR TITLE
Medbot-Bros Fixes / Adds Trait Callbacks

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
:cl: Medbotby
fix: Medbots do not idle while stationary because they see a hot babe across the medbay
fix: Medbots do not take you down with them upon deletion, resulting in no heals from the little buggers for the rest of the round
/:cl:

And a runtime I think.

### CODE REVIEW

**Comment by ATH1909** [comment]
Is "noheals" in the changelog intentionally one word?

Also, what with these PR names and commit names and all, I wonder who the first person to write medibot-focused curator-bait will be.

**Comment by ExcessiveUseOfCobblestone** [comment]
i dont get the ref

**Comment by nemvar** [comment]
What is the point of having the trait at all after this? I assumed it existed so multiple medbots can't heal you at the same time but it doesn't do that anymore.

**Comment by ExcessiveUseOfCobblestone** [comment]
I'm THIS CLOSE to making a proc that does REMOVE_TRAIT for the purpose of callback.

File: code/modules/mob/living/simple_animal/bot/medbot.dm [hunk 1]
```diff
@@ -305,7 +301,8 @@ GLOBAL_VAR(medibot_unique_id_gen)
 	if(emagged == 2) //Everyone needs our medicine. (Our medicine is toxins)
 		return TRUE
 
-	if(HAS_TRAIT(C,TRAIT_MEDIBOTCOMINGTHROUGH) && !HAS_TRAIT_FROM(C,TRAIT_MEDIBOTCOMINGTHROUGH,medibot_counter)) //someone is healing them already sweetie
+	if(HAS_TRAIT(C,TRAIT_MEDIBOTCOMINGTHROUGH)) //romance of the three medbots
+		REMOVE_TRAIT(C,TRAIT_MEDIBOTCOMINGTHROUGH,TRAIT_MEDIBOTCOMINGTHROUGH) //casus belli
```
Comment (msgerbs) [inline code/modules/mob/living/simple_animal/bot/medbot.dm]: These are funny, but surely we should have actual meaningful comments as to what this does?

Comment (msgerbs) [inline code/modules/mob/living/simple_animal/bot/medbot.dm]: These are funny, but surely we should have an actually meaningful comment as to what this does?

**Pull request merged**